dotnet/src/dotnetcore/GxClasses.Web/Middleware/GXRouting.cs

```diff
@@ -287,444 +290,442 @@ public Task ProcessRestRequest(HttpContext context)
 						result = Task.FromException(new PageNotFoundException(path));
 					}
 				}
-				context.CommitSession();
-
-				return result;
+				await context.CommitSessionAsync();
 			}
```

Comment:
There's a behavioral difference here. Previously it could return a faulted task with a `PageNotFoundException` if the route could not be handled. Now that case just returns a completed task.
It'd be simpler to just remove the `result` task variable, and maybe throw the `PageNotFoundException` when the path cannot be handled, and move the `await context.CommitSessionAsync()` to a finally block.